integration_tests/src/main/python/string_test.py

```diff
@@ -498,16 +498,26 @@ def test_re_replace():
                 'REGEXP_REPLACE(a, "TEST", NULL)'),
         conf=_regexp_conf)
 
-@allow_non_gpu('ProjectExec', 'RegExpReplace')
 def test_re_replace_backrefs():
-    gen = mk_str_gen('.{0,5}TEST[\ud720 A]{0,5}')
-    assert_gpu_fallback_collect(
+    gen = mk_str_gen('.{0,5}TEST[\ud720 A]{0,5}TEST')
+    assert_gpu_and_cpu_are_equal_collect(
         lambda spark: unary_op_df(spark, gen).selectExpr(
```

Comment:
Could you add the test case `'REGEXP_REPLACE(a, "(TEST)", "$1")'` since this fails for me locally